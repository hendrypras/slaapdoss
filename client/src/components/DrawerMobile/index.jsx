import PropTypes from 'prop-types';
import { Drawer } from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';

const DrawerMobile = ({ open, onClose, height = '100vh', padding = '0 1rem', display, children }) => (
  <Drawer
    sx={{
      height: '100vh',
      width: '100%',
      '& .MuiDrawer-paper': {
        height,
        padding,
        boxSizing: 'border-box',
        position: 'absolute',
        bottom: '0',
        borderRadius: '.5rem .5rem 0 0',
      },
    }}
    anchor="bottom"
    open={open}
    onClose={onClose}
  >
    <button
      type="button"
      onClick={onClose}
      aria-label="button-close"
      style={{
        position: 'absolute',
        right: '1rem',
        top: '0.6rem',
        fontSize: '2rem',
        backgroundColor: 'transparent',
        border: 'none',
        color: 'white',
        zIndex: '99',
        cursor: 'pointer',
        display,
      }}
    >
      <CloseIcon style={{ color: 'black' }} />
    </button>
    {children}
  </Drawer>
);

DrawerMobile.propTypes = {
  children: PropTypes.element,
  open: PropTypes.bool,
  height: PropTypes.string,
  padding: PropTypes.string,
  display: PropTypes.string,
  onClose: PropTypes.func,
};
export default DrawerMobile;
