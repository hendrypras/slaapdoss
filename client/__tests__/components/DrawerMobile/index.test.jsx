import { render, fireEvent } from '@utils/testHelper';
import DrawerMobile from '@components/DrawerMobile';

let isOpen = true;
let wrapper;

const mockProps = {
  open: isOpen,
  height: '100vh',
  onClose: () => {
    isOpen = false;
  },
};
beforeEach(() => {
  wrapper = render(<DrawerMobile open={mockProps.open} onClose={mockProps.onClose} />);
});

describe('DrawerMobile', () => {
  test('Should render correctly', () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-close');

    fireEvent.click(button);
    expect(isOpen).toBe(false);
    expect(getByTestId('drawer-mobile')).toBeInTheDocument();
  });
  test('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
