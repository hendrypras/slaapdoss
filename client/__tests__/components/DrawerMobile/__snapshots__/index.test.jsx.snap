// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerMobile should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDrawer-root MuiDrawer-modal MuiModal-root css-1t4p60r-MuiModal-root-MuiDrawer-root"
      data-testid="drawer-mobile"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorBottom css-9emuhu-MuiPaper-root-MuiDrawer-paper"
        tabindex="-1"
      >
        <button
          aria-label="button-close"
          data-testid="button-close"
          style="position: absolute; right: 1rem; top: 0.6rem; font-size: 2rem; background-color: transparent; color: white; z-index: 99; cursor: pointer;"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            style="color: black;"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </button>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
